src/modules/item/selectors.ts

```diff
@@ -94,35 +96,57 @@ export const getEntityByItemId = createSelector<RootState, Entity[], Record<stri
     }, {} as Record<string, Entity>)
 )
 
+const getItemSyncedStatus = (item: Item, entity: Entity | null) => {
+  let status = SyncStatus.UNSYNCED
+  if (!entity) {
+    status = SyncStatus.LOADING
+  } else if (areSynced(item, entity)) {
+    status = SyncStatus.SYNCED
+  }
+  return status
+}
+
+const getStatusForTPW = (item: Item, itemCuration: ItemCuration | null, entity: Entity): SyncStatus => {
+  let status: SyncStatus
+  if (!itemCuration) {
+    status = SyncStatus.UNPUBLISHED
+  } else if (itemCuration.status === CurationStatus.PENDING) return SyncStatus.UNDER_REVIEW
+  else {
+    status = getItemSyncedStatus(item, entity)
+  }
+  return status
+}
+
+const getStatusForDCL = (item: Item, collectionCuration: CollectionCuration | null, entity: Entity): SyncStatus => {
+  let status: SyncStatus
+  if (!item.isPublished) {
+    status = SyncStatus.UNPUBLISHED
+  } else if (!item.isApproved || (collectionCuration && collectionCuration.status === CurationStatus.PENDING)) {
+    status = SyncStatus.UNDER_REVIEW
+  } else {
+    status = getItemSyncedStatus(item, entity)
+  }
+  return status
+}
```

Comment:
following the current implementation in `master`

```ts
  if (!item.isPublished) {
    statusByItemId[item.id] = SyncStatus.UNPUBLISHED
  } else if (!item.isApproved) {
    statusByItemId[item.id] = SyncStatus.UNDER_REVIEW
  } else {
    const entity = entitiesByItemId[item.id]
    if (!entity) {
      statusByItemId[item.id] = SyncStatus.LOADING
    } else if (areSynced(item, entity)) {
      statusByItemId[item.id] = SyncStatus.SYNCED
    } else if (item.collectionId && curationsByCollectionId[item.collectionId]?.status === CurationStatus.PENDING) {
      statusByItemId[item.id] = SyncStatus.UNDER_REVIEW
    } else {
      statusByItemId[item.id] = SyncStatus.UNSYNCED
    }
  }
```

The original implementation used to favor LOADING over under_review for what I can see. Meaning that if `isApproved = true` and no entity was found the Status'd be LOADING but here could be UNDER_REVIEW If there's a collection status Pending right?

Reply:
True, that's right. In case there's no `entity`, the current code in `master` returns `LOADING` even after checking if there are curations or not for the collection (talking specifically about DCL collections).
I guess we gotta decide the precedence here, what's more relevant? Or, what should we check first? If it has an `entity` or a pending curation?